core/dbt/include/global_project/macros/adapters/apply_grants.sql

```diff
@@ -0,0 +1,50 @@
+{% macro get_show_grant_sql(relation) %}
+{{ return(adapter.dispatch("get_show_grant_sql", "dbt")(relation)) }}
+{% endmacro %}
+
+{% macro default__get_show_grant_sql(relation) %}
+show grants on {{ relation.type }} {{ relation }}
+{% endmacro %}
+
+{% macro get_grant_sql(relation, grant_config) %}
+{{ return(adapter.dispatch('get_grant_sql', 'dbt')(relation, grant_config)) }}
+{% endmacro %}
+
+{% macro default__get_grant_sql(relation, grant_config) %}
+    {% for privilege in grant_config.keys() %}
+        {% set grantee = grant_config[privilege] %}
+        grant {{ privilege }} on {{ relation.type }} {{ relation }} to {{ grantee | join(', ') }}
+    {% endfor %}
+{% endmacro %}
+
+{% macro get_revoke_sql(relation, grant_config) %}
+{{ return(adapter.dispatch("get_revoke_sql", "dbt")(relation, grant_config)) }}
+{% endmacro %}
+
+{% macro default__get_revoke_sql(relation, grant_config) %}
+    {% for privilege in grant_config.keys() %}
+        {% set grantee = grant_config[privilege] %}
+        revoke {{ privilege }} on {{ relation.type }} {{ relation }} from {{ grantee | join(', ') }}
+        where {{ grantee }} != {{ target.user }}
+    {% endfor %}
+{% endmacro %}
+
+{% macro apply_grants(relation, grant_config, should_revoke) %}
+{{ return(adapter.dispatch("apply_grants", "dbt")(relation, grant_config, should_revoke)) }}
+{% endmacro %}
+
+{% macro default__apply_grants(relation, grant_config, should_revoke=True) %}
+{% if grant_config %}
+    {% call statement('grants') %}
+        {% if should_revoke %}
+            {% set current_grants =  run_query(get_show_grant_sql(relation)) %}
+             {{ log(current_grants, info = true) }}
+            {% set diff_grants = { k: current_grants[k] for k in set(current_grants) - set(grant_config) } %}
```

Comment:
So I know inline python isn't really allowed in jinja without doing some extending or building out a new block of some form to allow it. so beyond building out a new single use function in impl or something like that is there a better way to try and grab the diffs?

Reply:
Yup, our options here are:
- writing this in Jinja, using `set` methods as you're trying here — I bet it's possible!
- adding a new Python context method to handle this logic for us (which could be defined as a [context member](https://github.com/dbt-labs/dbt-core/tree/main/core/dbt/context), or as a class method on the adapter)

Reply:
oh great, will continue playing around with trying to jinjafy the base python i'm trying to run was really digging into the postgres version of persist_docs as you mentioned